src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -1,52 +1,67 @@
 package io.pinecone;
 
 import io.pinecone.model.CreateIndexRequest;
-import org.asynchttpclient.AsyncHttpClient;
-import org.asynchttpclient.DefaultAsyncHttpClient;
-
+import okhttp3.*;
 import java.io.IOException;
+import static io.pinecone.utils.Constants.*;
 
 public class PineconeIndexOperationClient {
-    private AsyncHttpClient client;
-    private PineconeClientConfig clientConfig;
-    private String url;
+    private final OkHttpClient client;
+    private final PineconeClientConfig clientConfig;
+    private final String url;
 
-    PineconeIndexOperationClient(PineconeClientConfig clientConfig, AsyncHttpClient client) {
+    PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client) {
         this.clientConfig = clientConfig;
         this.client = client;
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this.url = BASE_URL_PREFIX + clientConfig.getEnvironment() + BASE_URL_SUFFIX;
     }
 
     public PineconeIndexOperationClient(PineconeClientConfig clientConfig) {
         this.clientConfig = clientConfig;
-        client = new DefaultAsyncHttpClient();
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this.client = new OkHttpClient();
+        this.url = BASE_URL_PREFIX + clientConfig.getEnvironment() + BASE_URL_SUFFIX;
     }
 
     public void deleteIndex(String indexName) throws IOException {
-        System.out.println("Sending delete index request:");
-        client.prepare("DELETE", url + indexName)
-                .setHeader("accept", "text/plain")
-                .setHeader("Api-Key", clientConfig.getApiKey())
-                .execute()
-                .toCompletableFuture()
-                .join();
-
-        client.close();
+        Request request = new Request.Builder()
+                .url(url + indexName)
+                .delete()
+                .addHeader(ACCEPT_HEADER, TEXT_PLAIN)
+                .addHeader(API_KEY, clientConfig.getApiKey())
+                .build();
+
+        try (Response response = client.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw new IOException(response.message());
+            }
+        } finally {
+            close(client);
```

Comment:
I don't like adding this close(client) here but keeping backwards compatibility in mind i.e. the older implementation used to close client within each method after the call was executed, I added it here. If I remove the close() call, it'll shift responsibility to the user to close the client after the usage.

Reply:
Great, that was the original thought of having the close() in the class and not calling it at the end of every method. I agree with all of the points you've made, so I'll remove the close(client) calls from each method and leave the close() as a part of the class.